iterative_verifier.go

```diff
@@ -239,10 +241,37 @@ func (v *IterativeVerifier) VerifyBeforeCutover() error {
 	return err
 }
 
+func (v *IterativeVerifier) reverifyUntilStoreIsSmallEnough(maxIterations int) error {
+	var timeToVerify time.Duration
+
+	for iteration := 0; iteration < maxIterations; iteration++ {
+		before := v.reverifyStore.RowCount
+		start := time.Now()
+
+		_, err := v.verifyStore("reverification_before_cutover", []MetricTag{{"iteration", string(iteration)}})
+		if err != nil {
+			return err
+		}
+
+		after := v.reverifyStore.RowCount
+		timeToVerify = time.Now().Sub(start)
+
+		if after >= before {
+			break
+		}
+	}
+
+	if v.MaxDowntime != 0 && timeToVerify > v.MaxDowntime {
+		return fmt.Errorf("Unable to verify data within max downtime duration (took %s)", timeToVerify)
```

Comment:
I find it confusing to call this "downtime". Doesn't this run before the cutover? So the shop isn't down yet and reverifying it more often doesn't increase the downtime.

Reply:
The rationale here is that if we ran our heuristic to completion, then we expect the cutover phase verification to take at least `timeToVerify` duration. This is why I chose `MaxDowntime`. Do you have any suggestions for more clarity?

Reply:
MaxReverificationTime?

Reply:
MaxExpectedDowntime?